research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,213 @@
+
+#### Slots
+
+*Example Table*
+
+| Slot Name | Description | Fallback Content |
+| --------- | ----------- | ---------------- |
+| `thumb` | Enables replacing the component's `thumb` part with custom markup. | A default thumb implementation is provided. |
+
+#### Host Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### Slotted Content Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### CSS Parts
+
+*Example Table*
+
+| Part Name | Description |
+| --------- | ----------- |
+| `thumb` | Enables styling the `thumb` part without requiring it to be completely replaced via its slot. |
+
+---
+
+## Implementation Plan
+
+### States and Interactions
+
+*List key component states, valid state transitions, and how interactions trigger state transitions.*
+
+- *Are states changed as a result of keyboard? mouse? touch?*
+- *Are there corresponding keyboard interactions for mouse/touch interactions?*
+- *Are some events triggered on key down vs key press?*
+
+*Example Table*
+
+| State Group | States | Initial State | Description | Interaction/Transition |
+| ----------- | ------ | ------------- | ----------- | ---------------------- |
+| `disabled` | `true`/`false` | `false` | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically. |
+
+### Accessibility
+
+*Consider the accessibility of the component.*
+
+#### Keyboard Navigation and Focus
+
+- *Arrow vs tabbing key behavior*
+- *Modifier key effects (e.g. shift, ctrl, meta)*
+- *Does this component use focus delegation?*
+
+*Example Table*
+
+| Key | Description |
+| --- | ----------- |
+| Up/Right  | Increments the value of the component by calling the `increment` method. If the `shift` modifier is pressed, increases by 10x the `step` value. |
+| Down/Left | Decrements the value of the component by calling the `decrement` method. If the `shift` modifier is pressed, decreases by 10x the `step` value. |
+
+#### Form Input
+
+*Does this component integrate with form submission, form validation, etc.?*
+
+#### Use with Assistive Technology
+
+*Are there any details about shadow dom, focus delegation, or aria attributes that need special attention?*
+
+### Globalization
+
+*Consider whether the component has any special globalization needs such as:*
+
+- *Special RTL handling*
+- *Swapping of internal icons/visuals*
+- *Localization*
+
+### Security
+
+*Are there any security implications surrounding the component?*
+
+### Performance
+
+*Are there any performance pitfalls or challenges with implementing the component?*
+
+### Dependencies
+
+*Will implementing the component require taking on any dependencies?*
+
+- *3rd party libraries*
+- *Upcoming standards we need to polyfill*
+- *Do any dependencies bring along an associated timeline?*
+
+### Test Plan
+
+*What is the plan for testing the component?*
```

Comment:
Would be good to include an example table to guide spec writers what to write here. The section may be redundant with some of the previous sections, so I'm not totally sure what to include.

| Scenario | Expectation |
|-|-|
| User clicks on enabled Checkbox with mouse | Checked state is toggled |
| User clicks on disabled Checkbox with mouse | Interaction is ignored, (is focus set to disabled checkbox?) |

Reply:
I'm not entirely sure either. I think what I had in my head was are there any testing details that are non-standard; special test considerations. I think most of what to test would be covered in the above sections. I'll see if I can clarify some wording here.